gcc/rust/lex/rust-lex.h

```diff
@@ -132,6 +141,19 @@ class Lexer
   Lexer (const char *filename, RAIIFile input, Linemap *linemap);
   ~Lexer ();
 
+  /**
+   * Lex the contents of a string instead of a file
+   */
+  static Lexer lex_string (std::string &input)
```

Comment:
To make this a little safer I think you should remove the '&'  on the parameter. Since this is no longer const we can no longer guarantee that there will be no side effects on that parameter anymore.